tests/unit/r2i/tensorrt/frame.cc

```diff
@@ -0,0 +1,185 @@
+/* Copyright (C) 2018-2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+ */
+
+#include <r2i/r2i.h>
+#include <r2i/tensorrt/frame.h>
+#include <fstream>
+#include <cstring>
+#include <memory>
+
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTest/MemoryLeakDetectorNewMacros.h>
+#include <CppUTest/MemoryLeakDetectorMallocMacros.h>
+#include <CppUTest/TestHarness.h>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#define FRAME_WIDTH  2
+#define FRAME_HEIGHT  2
+
+bool cudaMallocError = false;
+bool cudaMemCpyError = false;
+size_t cudaRequestedSize = 0;
+
+__host__ __cudart_builtin__ cudaError_t CUDARTAPI
```

Comment:
How does this compile? doesn't this need to be interpreted by NVCC?

Reply:
All of these flags are already included by the cuda headers, the actual function is meant to be run on the Host side and therefore compilable by gcc